app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,40 @@
 class PostsController < ApplicationController
-  def index; end
+  before_action :set_user
 
-  def show; end
+  def index
+    @posts = @user.posts
+  end
+
+  def show
+    @post = @user.posts.find(params[:id])
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.new(post_parameters)
+    @post.author_id = @user.id
+
+    if @post.save
+      flash[:notice] = 'The post is successfully submitted'
+      redirect_to user_post_path(@user, @post)
+    else
```

Comment:
- [ ] As stated in the requirement and as you've done in the previous review, kindly ensure that it is only the `current_user` that can create a new post. For example in line 20, instead of `user.id` it should be something like `current_user.id` etc.

<img width="652" alt="Untitled" src="https://github.com/AlaaAlsalem/blogaapp/assets/103386183/190b0b0b-fd5a-4df7-b272-4c07ab8f5385">


- [ ] As stated earlier, you will need to create a method called `current_user` in your `ApplicationController` to make current user data available to all controllers. It should return the first user from the database as stated in the database. ℹ️

<img width="449" alt="Untitled" src="https://github.com/AlaaAlsalem/blogaapp/assets/103386183/eae9a0ed-ce09-46d3-abb7-6cc40fd4e3fe">